src/appleseed/renderer/kernel/rendering/progressive/progressiveframerenderer.cpp

```diff
@@ -763,6 +763,8 @@ Dictionary ProgressiveFrameRendererFactory::get_params_metadata()
         "max_samples",
         Dictionary()
             .insert("type", "int")
+            .insert("default", "100000")
+            .insert("unlimited", "true")
             .insert("label", "Max Samples")
             .insert("help", "Maximum number of samples per pixel"));
```

Comment:
Hm, I think this needs to be changed to read `Maximum number of samples`, since this doesn't seem to be constraining maximum samples per pixel.

https://github.com/appleseedhq/appleseed/blob/master/src/appleseed/renderer/kernel/rendering/progressive/samplegeneratorjob.cpp#L127
The `ProgressiveFrameRenderer`'s termination condition hinges on a sample counter initialized with the total max sample count.

Reply:
That's correct, it currently controls the total number of samples. It's fine to expose those internal settings in appleseed.studio. In DCC apps we would probably expose "Max Samples per Pixel" instead since this is more useful to a user.

Retrospectively, maybe the actual control should have been "Max Average Samples per Pixel" both in the code, in appleseed.studio and in DCC apps because that makes the settings resolution-independent. It's not too late to change it :)

Reply:
Oh nice, and "Max Average Samples per Pixel" would cause termination if total samples / total pixels exceeds it. Since I'm already in here, I can rework the computation then squash the commits.

(I'll double check that "Max samples" isn't incorrectly exposed as "Max samples per pixel" elsewhere too, though I didn't think it was on my first pass here)

Reply:
> Oh nice, and "Max Average Samples per Pixel" would cause termination if total samples / total pixels exceeds it.

Exactly.

(I'll double check that "Max samples" isn't incorrectly exposed as "Max samples per pixel" elsewhere too, though I didn't think it was on my first pass here)

👍